extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallryeJwtDevServiceProcessor.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.smallrye.jwt.deployment;
+
+import static io.quarkus.smallrye.jwt.deployment.SmallRyeJwtProcessor.MP_JWT_VERIFY_KEY_LOCATION;
+
+import java.security.Key;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BooleanSupplier;
+import java.util.stream.Collectors;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.deployment.Feature;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.DevServicesResultBuildItem;
+import io.quarkus.deployment.builditem.LiveReloadBuildItem;
+import io.quarkus.runtime.LaunchMode;
+import io.smallrye.jwt.util.KeyUtils;
+
+public class SmallryeJwtDevServiceProcessor {
+
+    private static final String MP_JWT_VERIFY_PUBLIC_KEY = "mp.jwt.verify.publickey";
+
+    private static final String SMALLRYE_JWT_SIGN_KEY_LOCATION = "smallrye.jwt.sign.key.location";
+    private static final String SMALLRYE_JWT_SIGN_KEY = "smallrye.jwt.sign.key";
+    private static final String NONE = "NONE";
+
+    private static final int KEY_SIZE = 2048;
+
+    private static final Set<String> JWT_SIGN_KEY_PROPERTIES = Set.of(
+            MP_JWT_VERIFY_KEY_LOCATION,
+            MP_JWT_VERIFY_PUBLIC_KEY,
+            SMALLRYE_JWT_SIGN_KEY_LOCATION,
+            SMALLRYE_JWT_SIGN_KEY);
+
+    @BuildStep(onlyIf = { IsDevelopmentOrTest.class })
+    void generateSignKeys(BuildProducer<DevServicesResultBuildItem> devServices,
+            LiveReloadBuildItem liveReloadBuildItem) throws NoSuchAlgorithmException {
+
+        Config configProvider = ConfigProvider.getConfig();
+        Set<String> providedConfigs = JWT_SIGN_KEY_PROPERTIES
+                .stream()
+                .filter(prop -> !configProvider.getOptionalValue(prop, String.class)
+                        .orElse("")
+                        .isBlank())
+                .collect(Collectors.toSet());
+
+        if (!providedConfigs.isEmpty()) {
+            HashMap<String, String> config = new HashMap<>();
+            if (!providedConfigs.contains(SMALLRYE_JWT_SIGN_KEY)) {
+                config.put(SMALLRYE_JWT_SIGN_KEY, NONE);
+            }
+            if (!config.containsKey(MP_JWT_VERIFY_PUBLIC_KEY)) {
+                config.put(MP_JWT_VERIFY_PUBLIC_KEY, NONE);
+            }
+            devServices.produce(smallryeJwtDevServiceWith(config));
```

Comment:
If it is already in config, why you produce at as config properties again? Have you tried it and this is required? Please check.

Reply:
ok